spec/jobs/reports/monthly_account_reuse_report_spec.rb

```diff
@@ -0,0 +1,190 @@
+require 'rails_helper'
+require 'csv'
+
+RSpec.describe Reports::MonthlyAccountReuseReport do
+  subject(:report) { Reports::MonthlyAccountReuseReport.new }
+
+  let(:report_date) { Date.new(2021, 3, 1) }
+  let(:s3_public_reports_enabled) { true }
+  let(:s3_report_bucket_prefix) { 'reports-bucket' }
+  let(:s3_report_public_bucket_prefix) { 'public-reports-bucket' }
+  let(:s3_report_path) do
+    'int/monthly-account-reuse-report/2021/2021-03-01.monthly-account-reuse-report.json'
+  end
+
+  before do
+    allow(Identity::Hostdata).to receive(:env).and_return('int')
+    allow(Identity::Hostdata).to receive(:aws_account_id).and_return('1234')
+    allow(Identity::Hostdata).to receive(:aws_region).and_return('us-west-1')
+    allow(IdentityConfig.store).to receive(:s3_public_reports_enabled).
+      and_return(s3_public_reports_enabled)
+    allow(IdentityConfig.store).to receive(:s3_report_bucket_prefix).
+      and_return(s3_report_bucket_prefix)
+    allow(IdentityConfig.store).to receive(:s3_report_public_bucket_prefix).
+      and_return(s3_report_public_bucket_prefix)
+
+    Aws.config[:s3] = {
+      stub_responses: {
+        put_object: {},
+      },
+    }
+  end
+
+  describe '#perform' do
+    it 'uploads a file to S3 based on the report date' do
+      ['reports-bucket.1234-us-west-1', 'public-reports-bucket-int.1234-us-west-1'].each do |bucket|
```

Comment:
[ditto earlier comment about public bucket](https://github.com/18F/identity-idp/pull/9076/files#r1309213875), we just want the reports-bucket